controllers/grid/navigationMenus/form/PKPNavigationMenuItemsForm.php

```diff
@@ -180,19 +180,44 @@ public function execute(...$functionParams)
             $navigationMenuItem = $navigationMenuItemDao->newDataObject();
             $navigationMenuItem->setTitle($this->getData('title'), null);
         } else {
+            // The NMI will have localized title in the navigation_menu_item_settings table
+            // only if the user is defining one explicitely.
+            // This is not always the case for the default NMIs that, by default, get their titles
+            // from the titleLocaleKey setting of the NMI.
+
+            // Get the title that have been explicitely set in the settings tatble
+            $localizedTitlesFromDB = $navigationMenuItem->getTitle(null) ?? [];
+
+            // Get the set of all displayed titles, no matter the source. 
+            Services::get('navigationMenu')
+                ->setAllNMILocalizedTitles($navigationMenuItem);
+            $originalDisplayTitles = $navigationMenuItem->getTitle(null);
+
             $newTitles = [];
+
+            // Get the explicitely provided user titles
             $inputLocalizedTitles = $this->getData('title');
 
+            // Explicit define the localized title of the NMI.
             foreach ($inputLocalizedTitles as $locale => $inputTitle) {
-                $inputTitle = trim($inputTitle);
+                $trimmedTitle = trim($inputTitle);
 
-                // If the title is empty, set it to null for the locale
-                if ($inputTitle === '') {
+                // If empty, set to null (resets to default)
+                if ($trimmedTitle === '') {
                     $newTitles[$locale] = null;
                     continue;
                 }
 
-                $newTitles[$locale] = $inputTitle;
+                // Check if there is a display title for the NMI. If not for this locale, explicitely define one
+                if (!isset($originalDisplayTitles[$locale])) {
+                    $newTitles[$locale] = $trimmedTitle;
+                    continue;
+                }
+
+                // Update title if explicitly defined or changed from default
+                if (isset($localizedTitlesFromDB[$locale]) || $originalDisplayTitles[$locale] !== $trimmedTitle) {
+                    $newTitles[$locale] = $trimmedTitle;
+                }
```

Comment:
I think this can be left as:

```suggestion
                // Update title if explicitly defined or changed from default
                if (isset($localizedTitlesFromDB[$locale]) || ($originalDisplayTitles[$locale] ?? null) !== $trimmedTitle) {
                    $newTitles[$locale] = $trimmedTitle;
                }
```

Reply:
@jonasraoni are those two equivalent or you think that the suggested one incorporates the correct approach for this case?

Reply:
What do you say, to leave it in two different clauses for clarity (to an already "difficult" code segment to read)?